src/vs/editor/common/commonCodeEditor.ts

```diff
@@ -622,6 +622,10 @@ export abstract class CommonCodeEditor extends EventEmitter implements editorCom
 		return this.model.deltaDecorations(oldDecorations, newDecorations, this.id);
 	}
 
```

Comment:
Thanks for the explanation!
@aeschli how are you registering your folding end of line decorators? Can you provide a code pointer? Does it make sense to expose register editor decoration API as @nojvek is proposing here.